eras/shelley/impl/src/Cardano/Ledger/Shelley/TxWits.hs

```diff
@@ -215,12 +212,12 @@ pattern ShelleyTxWits {addrWits, scriptWits, bootWits} <-
             if null x then Nothing else Just (encodeWord ix <> enc x)
           l =
             catMaybes
-              [ encodeMapElement 0 encodeFoldable awits,
-                encodeMapElement 1 encodeFoldable scriptWitMap,
-                encodeMapElement 2 encodeFoldable bootstrapWits
+              [ encodeMapElement 0 toCBOR awits,
+                encodeMapElement 1 toCBOR (Map.elems scriptWitMap),
```

Comment:
The previous  version `encodeFoldable scriptWitMap` was also encoding only the  values of the map?

Reply:
That's right. Foldable instance for Map only concerns the elements, not the keys. So with this approach it is a bit more explicit that we discard the keys.

Reply:
Makes sense, thanks for the explanation